core/src/test/java/com/nuodb/migrator/backup/globalStore/GlobalStoreTest.java

```diff
@@ -0,0 +1,187 @@
+        column2.setTypeCode(BIGINT);
+        column2.setTypeName("BIGINT");
+        column2.setNullable(false);
+        column2.setPosition(1);
+        Sequence sequence2 = createSequence(null, "schema2", "test_table2", "id2");
+        sequence2.setName("sequence2");
+        column2.setSequence(sequence2);
+        schema2.addSequence(sequence2);
+        column2.getSequence().setGeneratedAlways(false);
+        table2.addColumn(column2);
+        schema2.addTable(table2);
+        gb.put(sequence2.getName(), sequence2.isGeneratedAlways());
+//        String alterStat2 =null;
+
+
+        Schema schema3 = new Schema("schema3");
+        Table table3 = new Table("test_table3");
+        Column column3 = table3.addColumn("id3");
+        column3.setTypeCode(BIGINT);
+        column3.setTypeName("BIGINT");
+        column3.setNullable(false);
+        column3.setPosition(1);
+        Sequence sequence3 = createSequence(null, "schema3", "test_table3", "id3");
+        sequence3.setName("sequence3");
+        column3.setSequence(sequence3);
+        schema3.addSequence(sequence3);
+//        column.getSequence().setGeneratedAlways(null);
+        table3.addColumn(column3);
+        schema3.addTable(table3);
+        gb.put(sequence3.getName(), sequence3.isGeneratedAlways());
+        String alterStat3 = "";
+
+
+        return new Object[][]{
+                {table, gb, dialect, column, alterStat},
+                {table2, gb, dialect, column2, null},
+                {table3, gb, dialect, column3, null}
+        };
+        // Arrange
+//        Dialect dialect = new NuoDBDialect();
+//        Schema schema = new Schema("schema1");
+//        Table table = new Table("test_table");
+//        Column column = table.addColumn("id");
+//        column.setTypeCode(BIGINT);
+//        column.setTypeName("BIGINT");
+//        column.setNullable(false);
+//        column.setPosition(1);
+//        Sequence sequence = createSequence(null, "schema1", "test_table", "id");
+//        column.setSequence(sequence);
+//        schema.addSequence(sequence);
+//        column.getSequence().setGeneratedAlways(true);
+//        table.addColumn(column);
+//        schema.addTable(table);
+//        when(mockedLoadTable.getTable()).thenReturn(table);
+//        globalStore.put("sequence_name", true);
+//        LoadTable tb = new LoadTable(null, table, null, 0);
+//        String alterScript = globalStore.alterScript(tb);
+////        String expected = "ALTER TABLE " + schema + "." + table + " MODIFY " + column + " " + column.getTypeName() + " " + "GENERATED ALWAYS AS IDENTITY";
+//        String expected = "ALTER TABLE  \"schema1\" . \"test_table\" MODIFY \"id\"  BIGINT GENERATED ALWAYS AS IDENTITY ";
+//      return new Object[][]{{dialect, expected, mockedLoadTable}};
+    }
+
+    @Test(dataProvider = "alterScript")
+    public void testGeneratedAlways(Table table, GlobalStore gb, Dialect dialect, Column col, String expected) {
```

Comment:
I'm confused about how this test is structured. We rely on the `@DataProvider`-annotated method to return an array of parameters that are injected in this method by the test framework.

It is very difficult to read and if the test fails, there will not be a descriptive test name associated with the test failure, but `testGeneratedAlways[1]` or whichever variant of the test failed. There are only three lines in this test which is being parameterized, so we are not actually reusing a lot of code, which is what this is parameterization facility is for.

I suggest separating this in three separate test cases and giving them each descriptive names. I don't think you need to use `@DataProvider` at all.